### ISSUE TITLE
Nan predictions

### ISSUE THREAD

**Issue by ghost** [mrebol/f2f-consistent-semantic-segmentation#3]
Hello, I have nan predictions after a certain amount of steps; the problem is solved if I re-initialize the weights after, e.g., 200 steps, but I would like to handle the states for the full video length. Do you have any idea about the reason? Also tried to do gradient clipping, but it doesn't work. The focal loss is going down very well and before turning into nan it is about 0.009; lowering the learning rate just delays the arrival of nan values.
